MTA-6826, 6827, 6828, 6829, 6830, 6831 CQA April work for rules development guide (#359)
migtools/mta-documentation: docs/topics/rules-development/about-rules.adoc

Reviewer: Here and throughout the whole doc set:

Rule sets spelling should be revisited.

Author: Sorry, but this is not a spelling issue? :) rule sets/rulesets both have correct spelling but they are formatted differently.

= The {ProductShortName} rules

[role="_abstract"]

> {ProductFullName} analyzes the application source code to find issues based on user-generated rule and default rule definitions. You can fix issues in an analysis report before migrating the code to a target platform. You can view the rule syntax information to create a new rule or a ruleset.

Reviewer: ```suggestion
{ProductFullName} analyzes the application source code to find issues based on user-generated and default rule definitions. You can fix issues in an analysis report before migrating the code to a target platform. You can view the rule syntax information to create a new rule or a ruleset.
```

'user-generated' and 'default' modify the same plural noun.

You can use rules or rulesets as input arguments in an analysis. You can perform a rule-based analysis in the {ProductShortName} CLI, user interface, or by using one of the IDE plugins for {ProductShortName} (Visual Studio Code or IntelliJ).

> The rules can be used by {mta-dl-plugin} to generate AI-assisted code resolutions. {mta-dl-plugin} uses the issue description and metadata in rules to form well-defined contexts. These contexts generate AI-assisted suggestions to resolve issues in the code that {ProductShortName} identified through an analysis.

Reviewer: How about this rewrite for the first sentence to remove passive voice?

Plus there are more instances of passive voice in this files and others, too. Is it intentional that we are fixing only this specific instance and not everything?

```suggestion
{mta-dl-plugin} can use the rules to generate AI-assisted code resolutions. It uses the issue description and metadata in rules to form well-defined contexts. These contexts generate AI-assisted suggestions to resolve issues in the code that {ProductShortName} identified through an analysis.
```
